DQM/CTPPS/plugins/TotemT2DQMSource.cc

```diff
@@ -0,0 +1,281 @@
+    static const unsigned int MINIMAL_TRIGGER = 3;
+
+    MonitorElement *leadingEdge = nullptr, *trailingEdge = nullptr, *timeOverTreshold = nullptr;
+
+    SectorPlots() = default;
+    SectorPlots(
+        DQMStore::IBooker& ibooker, unsigned int id, unsigned int nbinsx, unsigned int nbinsy, unsigned int windowsNum);
+  };
+
+  struct PlanePlots {
+    MonitorElement* digisMultiplicity = nullptr;
+    MonitorElement* rechitMultiplicity = nullptr;
+
+    PlanePlots() = default;
+    PlanePlots(DQMStore::IBooker& ibooker, unsigned int id, unsigned int nbinsx, unsigned int nbinsy);
+  };
+
+  std::unordered_map<unsigned int, SectorPlots> sectorPlots_;
+  std::unordered_map<unsigned int, PlanePlots> planePlots_;
+};
+
+TotemT2DQMSource::SectorPlots::SectorPlots(
+    DQMStore::IBooker& ibooker, unsigned int id, unsigned int nbinsx, unsigned int nbinsy, unsigned int windowsNum) {
+  std::string title, path;
+
+  TotemT2DetId(id).armName(path, TotemT2DetId::nPath);
+  ibooker.setCurrentFolder(path);
+
+  TotemT2DetId(id).armName(title, TotemT2DetId::nFull);
+
+  activePlanes = ibooker.book1D("active planes", title + " which planes are active;plane number", 8, -0.5, 7.5);
+
+  activePlanesCount = ibooker.book1D(
+      "number of active planes", title + " how many planes are active;number of active planes", 9, -0.5, 8.5);
+
+  triggerEmulator = ibooker.book2DD("trigger emulator",
+                                    title + " trigger emulator;arbitrary unit;arbitrary unit",
+                                    nbinsx,
+                                    0.,
+                                    nbinsx,
+                                    nbinsy,
+                                    0.,
+                                    nbinsy);
+  leadingEdge = ibooker.book1D(
+      "leading edge", title + " leading edge (DIGIs); leading edge (ns)", 25 * windowsNum, 0, 25 * windowsNum);
+  trailingEdge = ibooker.book1D(
+      "trailing edge", title + " trailing edge (DIGIs); trailing edge (ns)", 25 * windowsNum, 0, 25 * windowsNum);
+
+  timeOverTreshold = ibooker.book1D(
+      "time over threshold", title + " time over threshold (rechit);time over threshold (ns)", 250, -25, 100);
+}
+
+TotemT2DQMSource::PlanePlots::PlanePlots(DQMStore::IBooker& ibooker,
+                                         unsigned int id,
+                                         unsigned int nbinsx,
+                                         unsigned int nbinsy) {
+  std::string title, path;
+  TotemT2DetId(id).planeName(title, TotemT2DetId::nFull);
+  TotemT2DetId(id).planeName(path, TotemT2DetId::nPath);
+  ibooker.setCurrentFolder(path);
+
+  digisMultiplicity = ibooker.book2DD("digis multiplicity",
+                                      title + " digis multiplicity;arbitrary unit;arbitrary unit",
+                                      nbinsx,
+                                      0.,
+                                      nbinsx,
```

Comment:
This parameter should be a `double`
Then, for some reasonable histogram that reports integers in nbinsx bins, I imagine that it should better go from `-0.5` to `double(nbinsx)-0.5` (same for `nbinsy`)